crates/re_types_builder/src/codegen/python.rs

```diff
@@ -1491,6 +1491,156 @@ fn quote_arrow_support_from_obj(
     }
 }
 
+fn quote_argument_type_alias(
+    arg_type_fqname: &str,
+    class_fqname: &str,
+    objects: &Objects,
+    array: bool,
+) -> String {
+    let obj = &objects[arg_type_fqname];
+
+    let base = if let Some(delegate) = obj.delegate_datatype(objects) {
+        fqname_to_type(&delegate.fqname)
+    } else if arg_type_fqname == class_fqname {
+        // We're in the same namespace, so we can use the object name directly.
+        // (in fact we have to since we don't import ourselves)
+        obj.name.clone()
+    } else {
+        fqname_to_type(arg_type_fqname)
+    };
+
+    if array {
+        format!("{base}ArrayLike")
+    } else {
+        format!("{base}Like")
+    }
+}
+
+fn quote_init_argument_from_field(
+    field: &ObjectField,
+    objects: &Objects,
+    current_obj_fqname: &str,
+) -> String {
+    let type_annotation = if let Some(fqname) = field.typ.fqname() {
+        quote_argument_type_alias(fqname, current_obj_fqname, objects, field.typ.is_plural())
+    } else {
+        let type_annotation = quote_field_type_from_field(objects, field, false).0;
+        // Relax type annotation for numpy arrays.
+        if type_annotation.starts_with("npt.NDArray") {
+            "npt.ArrayLike".to_owned()
+        } else {
+            type_annotation
+        }
+    };
+
+    if field.is_nullable {
+        format!("{}: {} | None = None", field.name, type_annotation)
+    } else {
+        format!("{}: {}", field.name, type_annotation)
+    }
+}
+
+fn quote_init_method(obj: &Object, ext_class: &ExtensionClass, objects: &Objects) -> String {
```

Comment:
this code can surely golfed down so something more compact, but I already spent to much time on that aspect. If someone sees easy shortcuts that would be appreciated though.